sqlmesh/core/macros.py

```diff
@@ -171,13 +172,14 @@ def evaluate_macros(
 
             if isinstance(node, MacroVar):
                 changed = True
-                if node.name not in self.locals:
+                variables = self.locals.get(c.SQLMESH_VARS, {})
+                if node.name not in self.locals and node.name.lower() not in variables:
```

Comment:
why is lower() only called on the second one and not the first one?

Reply:
Seems like we rely on case sensitivity in locals. Some tests failed when I tried that. Investigating this is out of scope of this update.

Reply:
ok